### PR TITLE
chore: start adding missing end to end tests

### PR DESCRIPTION
**Pull request by cragwolfe**
✅ Sets the pattern for the comprehensive unittests in https://github.com/Unstructured-IO/unstructured-api-tools/issues/104 .

:white_check_mark:  Adds a pipeline test project with a root directory of `test_unstructured_api_tools/pipeline-test-project/`

:white_check_mark:  Generates web API's 
`test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/` from notebooks under 
`test_unstructured_api_tools/pipeline-test-project/pipeline-notebook` with `make generate-test-api`

:white_check_mark: Ensures the generated API's match expected in CI with `make api-check-test`

✅ Adds FastAPI client tests against the generated API's in `test_unstructured_api_tools/api/test_file_apis.py`

There are still many notebooks to add under `test_unstructured_api_tools/pipeline-test-project/pipeline-notebook`  and many more permutations to test per https://github.com/Unstructured-IO/unstructured-api-tools/issues/104 , but this is establishing the pattern.

### CODE REVIEW

**Comment by MthwRobinson** [comment]
@stamzid - Can you pick up review on this one?

**Comment by cragwolfe** [comment]
Cc @amanda103

File: test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/app.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#####################################################################
```
Comment (amanda103) [inline test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/app.py:1]: This file and the one below, they're just spinning up the test client so the test has something to interact with yes?

Comment (cragwolfe) [reply test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/app.py:1]: this is the webapp server for which the test client is spun up, and that its only purpose, yes

File: test_unstructured_api_tools/api/test_file_apis.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+FILENAME_FORMATS = {
+    FILE_A: "application/vnd.openxmlformats",
+    FILE_B: "image/jpeg",
+}
+
+P_INPUT_1_SINGLE = {"input_1": ["hi"]}
+P_INPUT_1_MULTI = {"input_1": ["hi", "water is better than ice"]}
+P_INPUT_2_SINGLE = {"input_2": ["hello"]}
+P_INPUT_2_MULTI = {"input_2": ["hello", "earth is better than mars"]}
+P_INPUT_1_AND_2_MULTI = {"input_2": ["hello", "earth is better than mars"], "input_1": ["hi"]}
```
Comment (amanda103) [inline test_unstructured_api_tools/api/test_file_apis.py:26]: I'm a little confused in following the api call, about how these params are actually being used. Can you point me to the place where these inputs change our expected response?

File: test_unstructured_api_tools/api/test_file_apis.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        ([FILE_B], P_INPUT_1_AND_2_MULTI, JSON, 200),
+        ([FILE_A, FILE_B], P_INPUT_1_AND_2_MULTI, JSON, 200),
+        ([FILE_A, FILE_B], P_INPUT_1_AND_2_MULTI, MIXED, 200),
+        # json returned though mixed requested (maybe not a bug for 1 file?)
+        pytest.param([FILE_A], P_INPUT_1_MULTI, MIXED, 200, marks=pytest.mark.xfail),
+        # json returned though csv requested
+        pytest.param([FILE_B], P_INPUT_1_AND_2_MULTI, TEXT_CSV, 200, marks=pytest.mark.xfail),
+    ],
+)
+def test_process_file_1(test_files, test_params, test_type_header, expected_status):
```
Comment (amanda103) [inline test_unstructured_api_tools/api/test_file_apis.py:82]: Probably outside of the scope of this PR/body of work but we'll likely want some non-200 response test cases too!

**Comment by amanda103** [comment]
Okay these tests look good for a baseline happy path! I can see the new tests are passing in the GHA check, but maybe we can pair on running these new tests locally and I can smash the approval button @cragwolfe ?

**Comment by cragwolfe** [comment]
> Okay these tests look good for a baseline happy path! I can see the new tests are passing in the GHA check, but maybe we can pair on running these new tests locally and I can smash the approval button @cragwolfe ?

Sounds good, let's do that on Monday!

**Review by amanda103 — APPROVED** [review]
Paired synchronously, LGTM!

**Pull request merged**